edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Edgelet.Test/ModuleManagementHttpClientTest.cs

```diff
@@ -348,5 +377,18 @@ public static IEnumerable<object[]> GetVersionMap()
             List<object[]> versionMap = Versions.SelectMany(x => Versions.Select(y => new object[] { x, y })).ToList();
             return versionMap;
         }
+
+        public static IEnumerable<object[]> GetInvalidDateTime()
+        {
+            var invalidDateTimes = new List<(string Since, string Until)>()
+            {
+                ($"2021-11-01T12:07:0Z", "2021-11-01T12:07:44Z"),
+                ("2021-11-01T12:07:32Z", "2021-11-01T12:07:0Z"),
+                ("random1", "random2"),
+            };
+
+            var incorrect = GetVersionMap().SelectMany(x => invalidDateTimes.Select(y => new object[] { x[0], x[1], y.Since, y.Until }));
```

Comment:
I tried to understand what structure this returns, but it confused me. Can we give this an explicit type? Or a comment explaining the structure?

Reply:
On first glance I thought it was an object like was being created in the inner select:
`new object[] { x[0], x[1], y.Since, y.Until }`

Looking harder I didn't think it was this. But based on method params, it is actually that simple. The `SelectMany` is what confused me. But I understand it now.